### ISSUE TITLE
automatic tune sample to fit pattern length

### ISSUE THREAD

**Issue by gherkins** [polyend/TrackerBetaTesting#1469]
<sup>searched for it, didn't find it, still think it's a good idea...</sup>

Imagine this:

* go to instrument parameters
* hold down shift 
* "tune" becomes "tune to pattern length"
* click and have the tuning/finetuning settings adjusted to make the sample fit the pattern length

*instant seamless looping* 🎉

Now that would be really nice, wouldn't it?

**Comment by breskar** [comment]
Yes please, the good old sync sample length to a number of steps.

**Comment by theorize999** [comment]
please please please introduce this, It really is a pain in the rear doing this manually, and I'd rather see this fit to xx steps than pattern length personally since xx steps could be set to pattern length. This would be a huge improvement!

**Comment by breskar** [comment]
as #1466  Sample Tune to Lines

**Comment by Djeroek** [comment]
+1 Would be great to have syncing options to X amount of lines for samples, auto transposing and finetuning until it fits.

**Comment by the-invisible-bears** [comment]
syncing sample length to pattern length would be a godsend.  I use it all the time in my software tracker and it's the one feature I was a little bummed to find out didn't exist on the Polyend Tracker. Please, bring this!

**Comment by Beddings** [comment]
Yes, please, a thousand times.

**Comment by NoSitRecords** [comment]
Yesssssssss

**Comment by shawnroos** [comment]
➕ 💯
